From b7dbf9fc6407ca4ac0bc1b1124cefedf570bf851 Mon Sep 17 00:00:00 2001
From: Liu Hao <lh_mouse@126.com>
Date: Wed, 2 May 2018 10:48:01 +0800
Subject: [PATCH 1/2] gendef: Silence warnings about casting function pointers
 on GCC 8.

GCC 8 warns about casting between incompatible pointer to functions.
This can be worked around by casting the source pointer to an integer
type before to the destination pointer type.

Signed-off-by: Liu Hao <lh_mouse@126.com>
(cherry picked from commit ebd3e57603879490d0ff187c25382703e8ccad2a)
Signed-off-by: Jonathan Yong <10walls@gmail.com>
---
 mingw-w64-tools/gendef/src/fsredir.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mingw-w64-tools/gendef/src/fsredir.c b/mingw-w64-tools/gendef/src/fsredir.c
index 36afd838..1e848c08 100644
--- a/mingw-w64-tools/gendef/src/fsredir.c
+++ b/mingw-w64-tools/gendef/src/fsredir.c
@@ -41,8 +41,8 @@ void doredirect(const int redir) {
       fprintf(stderr, "kernel32.dll failed to load, failed to disable FS redirection.\n");
       return;
     }
-    redirectorfunction = (redirector)GetProcAddress(kernel32handle, "Wow64DisableWow64FsRedirection");
-    revertorfunction = (revertor)GetProcAddress(kernel32handle, "Wow64RevertWow64FsRedirection");
+    redirectorfunction = (redirector)(INT_PTR)GetProcAddress(kernel32handle, "Wow64DisableWow64FsRedirection");
+    revertorfunction = (revertor)(INT_PTR)GetProcAddress(kernel32handle, "Wow64RevertWow64FsRedirection");
     if (!redirectorfunction || ! revertorfunction) {
       FreeLibrary(kernel32handle);
       fprintf(stderr, "Wow64DisableWow64FsRedirection or Wow64RevertWow64FsRedirection functions missing.\n");
-- 
2.18.0.windows.1

